lib/wasi/src/state/builder.rs

```diff
@@ -0,0 +1,293 @@
+//! Builder code for [`WasiState`]
+
+use crate::state::{WasiFs, WasiState};
+use std::path::{Path, PathBuf};
+
+/// Creates an empty [`WasiStateBuilder`].
+pub fn create_wasi_state() -> WasiStateBuilder {
+    WasiStateBuilder::default()
+}
+
+/// Type for building an instance of [`WasiState`]
+///
+/// Usage:
+///
+/// ```
+/// # use wasmer_wasi::state::create_wasi_state;
+/// create_wasi_state()
+///    .env(b"HOME", "/home/home".to_string())
+///    .arg("--help")
+///    .envs({ let mut hm = std::collections::HashMap::new();
+///            hm.insert("COLOR_OUTPUT", "TRUE");
+///            hm.insert("PATH", "/usr/bin");
+///            hm
+///          })
+///    .args(&["--verbose", "list"])
+///    .preopen_dir("src")
+///    .map_dir("dot", ".")
+///    .build()
+///    .unwrap();
+/// ```
```

Comment:
I was thinking that perhaps this API might be simpler? (removing `crate_wasi_state` and renaming `WasiStateBuilder` to `WASICommand` )

```
use wasmer_wasi::WASICommand;

WASICommand::new("")
   .env(b"HOME", "/home/home".to_string())
   .arg("--help")
   .envs({ let mut hm = std::collections::HashMap::new();
           hm.insert("COLOR_OUTPUT", "TRUE");
           hm.insert("PATH", "/usr/bin");
           hm
         })
   .args(&["--verbose", "list"])
   .preopen_dir("src")
   .map_dir("dot", ".")
   .build()
   .unwrap();
```